pymatgen/transformations/standard_transformations.py

```diff
@@ -716,7 +721,9 @@ def apply_transformation(self, structure):
                 old_occ = sp[k]
                 new_occ = float(
                     Fraction(old_occ).limit_denominator(self.max_denominator))
-                if round(abs(old_occ - new_occ), 6) > (
+                if fix_denominator: 
+                        new_occ = round(old_occ*self.max_denominator)/self.max_denominator
```

Comment:
This is a bad implementation. What if the rounding is out of tolerance and fix_denominator is true? That will round the occupancies no matter what and by pass the check in the elif statement.

Reply:
It is unlikely someone wants to use a fixed_denominator only within a given tolerance, but I changed this now so there is no bypass of the tolerance anymore. Hope I fixed the bad implementation that way.